scripts/licenses/licenses/library_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package licenses
+
+import (
+	"go/build"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestLibaries(t *testing.T) {
+	for _, test := range []struct {
+		desc       string
+		importPath string
+		workingDir string
+		importMode build.ImportMode
+		wantLibs   []string
+	}{
+		{
+			desc:       "Detects direct dependency",
+			importPath: "github.com/google/trillian/scripts/licenses/licenses/testdata/direct",
+			wantLibs: []string{
+				"github.com/google/trillian/scripts/licenses/licenses/testdata/indirect",
+			},
+		},
+		{
+			desc:       "Detects transitive dependency",
+			importPath: "github.com/google/trillian/scripts/licenses/licenses/testdata",
+			wantLibs: []string{
+				"github.com/google/trillian/scripts/licenses/licenses/testdata/direct",
+				"github.com/google/trillian/scripts/licenses/licenses/testdata/indirect",
+			},
+		},
```

Comment:
be nice to have a test case with no overlap (e.g. "github.com/google/trillian" and "golang.org/x/crypto")

Reply:
Unfortunately, adding any external package to the import chain for this test could cause the test to break at any time. This is because changes to the imports in that external package (or any of its dependencies) would change the test output.